### PR TITLE
Documentation for sympy.liealgebras

### LINKED ISSUE #9171: Sphinx docs for sympy.liealgebras
As of now, docs for Lie Algebra Module is not there.

### PR DESCRIPTION
**Pull request by tokencolour**
Autogenerated Sphinx documentation for sympy.liealgebras
Fixes #9171

### CODE REVIEW

**Comment by moorepants** [comment]
Please check to make sure it renders properly when you build the docs on your computer.

**Comment by tokencolour** [comment]
I built the docs and checked. It renders as I expect it to be. I have no idea why the checks have failed. Please help me with this.

File: doc/src/modules/liealgebras/index.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+========
+Lie Algebra Module
+========
```
Comment (leosartaj) [inline doc/src/modules/liealgebras/index.rst]: This is too short. It should be atleast as long as the text.

File: doc/src/modules/liealgebras/index.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+.. autoclass:: sympy.liealgebras.cartan_type.CartanType_generator
+    :members:
+
+.. autoclass:: sympy.liealgebras.cartan_type.Standard_Cartan
+    :members:
+
+.. autofunction:: sympy.liealgebras.dynkin_diagram.DynkinDiagram
+
+.. autofunction:: sympy.liealgebras.cartan_matrix.CartanMatrix
+
```
Comment (leosartaj) [inline doc/src/modules/liealgebras/index.rst]: Maybe you can remove this blank line.

Comment (tokencolour) [inline doc/src/modules/liealgebras/index.rst]: @leosartaj I have made the changes that you suggested. Thanks!

**Comment by leosartaj** [comment]
You probably need to add ``liealgebra/index.rst`` somewhere in ``doc/src/modules/index.rst``. I suggest try running the test related to documentation before committing. You can also try building the docs and checking for any warning or errors.

**Comment by tokencolour** [comment]
@leosartaj I have added the changes that you suggested. Thank you!

**Comment by leosartaj** [comment]
Maybe you can squash the commits into just one commit.

**Comment by leosartaj** [comment]
+1 to merge.

**Comment by tokencolour** [comment]
So as to squash commits, do I have to use `git rebase --interactive`?

**Comment by leosartaj** [comment]
Have a look [here](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html).

**Comment by tokencolour** [comment]
I squashed the commits into just one commit and then tried to push the commit. It showed the following error
```
To https://github.com/tokencolour/sympy.git
 ! [rejected]        lie_doc -> lie_doc (non-fast-forward)
error: failed to push some refs to 'https://github.com/tokencolour/sympy.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```
Henceforth, I updated the master branch by `git pull origin master` after switching into the master.
Then I switched to lie_doc branch (which is the branch I was working in while editing the docs) and merged with master using `git merge master`. 
Now, when I try `git push tokencolour lie_doc`, it shows the same error:
```
To https://github.com/tokencolour/sympy.git
 ! [rejected]        lie_doc -> lie_doc (non-fast-forward)
error: failed to push some refs to 'https://github.com/tokencolour/sympy.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

**Comment by moorepants** [comment]
It is not important to squash the commits. You can reset back or abort your squash.

**Comment by moorepants** [comment]
Note that if you change the git history you will have to force a push. There is little reason to rebase, see the tips here: https://github.com/sympy/sympy/wiki/Development-workflow#rebasing

@leosartaj Please avoid asking people to squash and rebase, especially new contributors. It adds an unnecessary overhead.

**Comment by tokencolour** [comment]
I have undone the merge and squash. Thank you for the help, @moorepants. Is there anything left to be done on my part?

**Comment by moorepants** [comment]
I think it is good to merge. Thanks!

**Comment by leosartaj** [comment]
> Please avoid asking people to squash and rebase, especially new contributors. It adds an unnecessary overhead.

@moorepants I will keep that mind.

Sorry for the delay. I have been having connectivity issues the past few days.

Thanks @tokencolour!

**Pull request merged**